cq-image/Makefile

```diff
@@ -0,0 +1,68 @@
+.PHONY := lint check fix docker_lint jinja_lint python_lint_check shell_lint yaml_lint jinja_format_check python_format_check shell_format_check yaml_format_check jinja_format_fix python_lint_fix python_format_fix shell_format_fix yaml_format_fix help
+
+all: lint check
+
+lint: docker_lint jinja_lint python_lint_check shell_lint yaml_lint
+
+check: jinja_format_check python_format_check shell_format_check yaml_format_check
+
+fix: jinja_format_fix python_lint_fix python_format_fix shell_format_fix yaml_format_fix
+
+help: 
+	@echo "make all: lint check"; \
+	echo "make lint: docker_lint jinja_lint python_lint_check shell_lint yaml_lint"; \
+	echo "make check: jinja_format_check python_format_check shell_format_check yaml_format_check"; \
+	echo "make fix: jinja_format_fix python_lint_fix python_format_fix shell_format_fix yaml_format_fix";
+
+docker_lint:
+	-for dockerfile in $$(bash -c 'find /code -name "Dockerfile"'); do \
+		echo "Checking $$dockerfile:"; \
+		hadolint $$dockerfile; \
+	done
+
+jinja_lint:
+	-for jinjafile in $$(bash -c 'find /code -name "*.j2"'); do \
+		djlint $$jinjafile --lint; \
+    done
+
+jinja_format_check:
+	-djlint /code --check --extension=j2
```

Comment:
I assume that the `-` is to ignore errors. Why is this being used? Why is it used only on every other command? This smells bad and maybe there is a better way.

Reply:
The `-` at the beginning of the command tells `make` to ignore non-zero exit codes, which is notably not the same as ignoring or suppressing errors in the command. Code quality tools will sometimes give non-zero exit codes when errors are found. With this structure, the top-level `make` commands `all`, `lint`, and `check` will run through the whole suite of tools, rather than forcing you to (for example) fix all your `Dockerfile` lint errors before being alerted to the `yaml` lint errors.

I notably did not include this in the `fix` commands, as running `make fix` acknowledges that you're writing to the codebase, which means that you should be responsible for fixing the errors that you find before those changes are committed.